File: src/mod10_design_pattern/lesson6_decorator/java/Main.java
package mod10_design_pattern.lesson6_decorator.java;

public class Main {
    public static void main(String[] args) {

        Wand basicWand = new BasicWand();
        Wand wandWithLumos= new LumosWand(basicWand);
        Wand wandWithFire= new FireWand(basicWand);

        System.out.println(basicWand.castSpell());
        System.out.println(wandWithLumos.castSpell());
        System.out.println(wandWithFire.castSpell());

    }
}

File: src/mod10_design_pattern/lesson6_decorator/kotlin/Main.kt
package mod10_design_pattern.lesson6_decorator.kotlin

fun main() {

    val basicWand = PatronusWand()//1
    val fireWand = FireWand(basicWand)//2
    val lumosWand = LumosWand(fireWand)//3

    println(basicWand.castSpell())//Expecto Patronum!
    println(fireWand.castSpell())//Expecto Patronum! Expelliarmus!
    println(lumosWand.castSpell())//Expecto Patronum! Expelliarmus! Lumos!
}